source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -602,13 +593,43 @@ class Filter : public Http::StreamFilter, private Filters::Common::Lua::LuaLogga
       UNREFERENCED_PARAMETER(host_and_strict);
     }
     void clearRouteCache() override {}
+    const ProtobufWkt::Struct& filterContext() const override { return parent_.filterContext(); }
 
     Filter& parent_;
     Http::StreamEncoderFilterCallbacks* callbacks_{};
   };
 
   using StreamHandleRef = Filters::Common::Lua::LuaDeathRef<StreamHandleWrapper>;
 
+  PerLuaCodeSetup* getPerLuaCodeSetup() {
+    if (decoder_callbacks_.callbacks_ != nullptr) {
+      per_route_config_ = Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(
+          decoder_callbacks_.callbacks_);
+    }
+
+    if (per_route_config_ != nullptr) {
+      // The filter is disabled by the route configuration explicitly.
+      if (per_route_config_->disabled()) {
+        return nullptr;
+      }
+      // The filter should execute the script specified by the script name if exist.
+      if (!per_route_config_->name().empty()) {
+        return config_->perLuaCodeSetup(per_route_config_->name());
+      }
+      // The filter should execute the script specified by the inline code if exist.
+      if (auto inline_code = per_route_config_->perLuaCodeSetup(); inline_code != nullptr) {
+        return inline_code;
+      }
+    }
+
+    return config_->perLuaCodeSetup();
+  }
+
+  const ProtobufWkt::Struct& filterContext() const {
+    return per_route_config_ == nullptr ? ProtobufWkt::Struct::default_instance()
```

Comment:
perf-nit: seems a bit wasteful to always pass the default_instance in cases where the filter_context field wasn't set.
I'm not sure what are the performance implications, and that's why I'd leave it up to the code-owners (you) to decide.

Reply:
The ProtobufWkt::Struct::default_instance() will return a const refrerence to a global variable. Basically, it won't bring any performance penality.

Reply:
> The ProtobufWkt::Struct::default_instance() will return a const refrerence to a global variable. Basically, it won't bring any performance penality.

I was refering to the call to `luaFilterContext` that I don't understand when it will to be invoked.
If you think that's ok, or that the overhead will be minimal, SGTM.

Reply:
This should only be invoked if the provided lua script requests this data, and it isn't configured. The alternative is to make this `optional` and return a `nil` in lua, but I think doing it this way is fine.